### DISCUSSION TITLE
Request matching doesn't seem to match with what the docs say.

### DISCUSSION THREAD

**Discussion by alveshelio** [mswjs/msw#1513] [Q&A] [answered]
Hi,

I'm not sure I understand how request matching works. I've looked at https://mswjs.io/docs/basics/request-matching 
And it has `Exact request URL`, `Path`, `Path with wildcard` and `Path with parameters`

I have these two URLs: `/api/competition-configurations/:id` (Path with parameters) and `/api/competition-configurations/default` (Exact request URL)

I have a service with this url `/api/competition-configurations/default` so I'd expect that the handler with the route `/api/competition-configurations/default` would be used, instead the handler with the URL `/api/competition-configurations/:id` is selected instead. 
If I change the order of the handlers and the one with the URL `/api/competition-configurations/default` is before the one with `/api/competition-configurations/:id` then the first one (`/default`) is picked.

What's even stranger is that if I do a one time override, then the route `/api/competition-configurations/default` gets picked.

Am I missing something? Does the order of the handlers really maters?

**Comment by kettanaito** [comment, marked as answer]
Hey, @alveshelio. 

I recommend reading this discussion: https://github.com/mswjs/msw/discussions/1426. I go into detail why this happens and how you should structure handlers for such ambiguous request paths.

**Comment by alveshelio** [comment]
Hi @kettanaito,

Thank you very much for the reply. After reading about the discussion it makes sense. I'll keep that in mind when creating the handlers. 

I think this is something that should be in the docs to help people understanding the order of the handlers.

**Reply by kettanaito** [reply]
Certainly! I'm working on the new docs, but will add this in the request matching section. Thanks.
